qemu/tests/balloon_check.py

```diff
@@ -89,6 +90,54 @@ def memory_check(self, step, ballooned_mem):
                 raise exceptions.TestFail("Balloon test failed %s" % step)
         return (mmem, gmem)
 
+    def memory_stats_check(self, keyname):
+        """
+        Check memory statistics via balloon service.
+
+        :param keyname: key name of the output of the 'qom-get' property.
+        :return: if memory statistics can be got via balloon service, reture True,
+                 else return False.
+        :rtype: bool
+        """
+        check_mem_ratio = float(self.params.get("check_mem_ratio", 0.1))
+        sleep_time = int(self.params.get("polling_sleep_time", 20))
+        base_path = self.params.get("base_path", "/machine/peripheral/")
+        device = self.params.get("balloon", "balloon0")
+        device_path = os.path.join(base_path, device)
+        polling_interval = int(self.params.get("polling_interval", 2))
+        set_balloon_property = self.params.get("set_balloon_property",
+                                               "guest-stats-polling-interval")
+        get_balloon_property = self.params.get("get_balloon_property", "guest-stats")
+
+        error_context.context("Get memory from guest", logging.info)
+        if keyname == "stat-free-memory":
+            guest_mem = self.get_guest_free_mem(self.vm)
+        elif keyname == "stat-total-memory":
+            guest_mem = self.get_vm_mem(self.vm)
+
+        error_context.context("Get memory from qmp", logging.info)
+        self.vm.monitor.qom_set(device_path, set_balloon_property, polling_interval)
+        time.sleep(sleep_time)
+        get_polling_output = self.vm.monitor.qom_get(device_path,
+                                                     get_balloon_property)
+        stat_memory_qmp = get_polling_output['stats'][keyname]
+
+        if stat_memory_qmp == 18446744073709551615:
```

Comment:
This number means balloon stat is not available which usually because balloon service is not running. 
For why it is this particular number please refer to bug1517055